state-chain/primitives/src/chains/assets.rs

```diff
@@ -302,16 +368,16 @@ macro_rules! assets {
 assets!(pub enum Asset {
 	// 0 is reserved for particular cross chain messaging scenarios where we want to pass
 	// through a message without making a swap.
-	(eth, Ethereum) => {
-		(Eth, "eth") = 1u32 (GAS_ASSET),
-		(Flip, "flip") = 2u32,
-		(Usdc, "usdc") = 3u32,
+	(eth, Ethereum, "Ethereum") => {
```

Comment:
This could also just be stringify!($chain) ?

Reply:
Yeah I tried, in this case it is impossible because stringify!(...) is an expression (Kind of), but the serde attribute needs a string literal. I think it is about the ordering of macro eval.